Pilot/pilotCommands.py

```diff
@@ -256,14 +242,80 @@ def _installDIRAC(self):
       except (IndexError, ValueError):
         continue
     # At this point self.pp.installEnv should contain all content of bashrc, sourced "on top" of (maybe) os.environ
-    self.pp.diracInstalled = True
+
+  def _installDIRACpy3(self):
+    """ Install python3 version of DIRAC client
+    """
+
+    # 1. Download DIRACOS
+    # curl -O -L https://github.com/DIRACGrid/DIRACOS2/releases/latest/download/DIRACOS-Linux-$(uname -m).sh
+    try:
+      machine = os.uname().machine  # py3
+    except AttributeError:
+      machine = os.uname()[4]  # py2
+
+    # FIXME: we should have a (set of) different location(s)
+    if not retrieveUrlTimeout(
+        "https://github.com/DIRACGrid/DIRACOS2/releases/latest/download/DIRACOS-Linux-%s.sh" % machine,
+        "DIRACOS-Linux-%s.sh" % machine,
+        self.log
+    ):
+      self.exitWithError(1)
+
+    # 2. bash DIRACOS-Linux-$(uname -m).sh
+    retCode, _ = self.executeAndGetOutput("bash DIRACOS-Linux-%s.sh" % machine, self.pp.installEnv)
+    if retCode:
+      self.log.error("Could not install DIRACOS [ERROR %d]" % retCode)
+      self.exitWithError(retCode)
+
+    # 3. rm DIRACOS-Linux-$(uname -m).sh
+    os.remove("DIRACOS-Linux-%s.sh" % machine)
+
+    # 4. source diracos/diracosrc then add its content to installEnv
+    retCode, output = self.executeAndGetOutput('bash -c "source diracos/diracosrc && env"', self.pp.installEnv)
+    if retCode:
+      self.log.error("Could not parse the diracos/diracosrc file [ERROR %d]" % retCode)
+      self.exitWithError(retCode)
+    for line in output.split('\n'):
+      try:
+        var, value = [vx.strip() for vx in line.split('=', 1)]
+        if var == '_' or 'SSH' in var or '{' in value or '}' in value:  # Avoiding useless/confusing stuff
+          continue
+        self.pp.installEnv[var] = value
+      except (IndexError, ValueError):
+        continue
+
+    # 5. pip install DIRAC==version
+    # FIXME: also for extensions
+
+    if self.pp.modules:
+      # https://github.com/$DIRAC_test_repo/DIRAC.git:::DIRAC:::$DIRAC_test_branch
+      url, _, branch = self.pp.modules.split(":::")
+      # pip install git+https://github.com/fstagni/DIRAC@v7r2-fixes33
+      retCode, output = self.executeAndGetOutput(
+          'pip install git+%s@%s' % (url.replace('.git', ''), branch),
```

Comment:
I think this one is correct but it should be checked
```suggestion
          'pip install git+%s@%s[pilot]' % (url.replace('.git', ''), branch),
```

Reply:
There is also no need to replace the `.git`

Reply:
This one does not work:

```
(base) root@9af1056655b8:/home# pip install git+https://github.com/fstagni/DIRAC.git@v7r2-fixes33[pilot]
Collecting git+https://github.com/fstagni/DIRAC.git@v7r2-fixes33[pilot]
  Cloning https://github.com/fstagni/DIRAC.git (to revision v7r2-fixes33[pilot]) to /tmp/pip-req-build-l3422m0w
  Running command git clone -q https://github.com/fstagni/DIRAC.git /tmp/pip-req-build-l3422m0w
  WARNING: Did not find branch or tag 'v7r2-fixes33[pilot]', assuming revision or ref.
  Running command git checkout -q 'v7r2-fixes33[pilot]'
  error: pathspec 'v7r2-fixes33[pilot]' did not match any file(s) known to git
WARNING: Discarding git+https://github.com/fstagni/DIRAC.git@v7r2-fixes33[pilot]. Command errored out with exit status 1: git checkout -q 'v7r2-fixes33[pilot]' Check the logs for full command output.
ERROR: Command errored out with exit status 1: git checkout -q 'v7r2-fixes33[pilot]' Check the logs for full command output.
```

This does:
```
(base) root@9af1056655b8:/home# pip install git+https://github.com/fstagni/DIRAC.git@v7r2-fixes33       
Collecting git+https://github.com/fstagni/DIRAC.git@v7r2-fixes33
  Cloning https://github.com/fstagni/DIRAC.git (to revision v7r2-fixes33) to /tmp/pip-req-build-y7f2i51w
  Running command git clone -q https://github.com/fstagni/DIRAC.git /tmp/pip-req-build-y7f2i51w
  Running command git checkout -b v7r2-fixes33 --track origin/v7r2-fixes33
  Switched to a new branch 'v7r2-fixes33'
  Branch 'v7r2-fixes33' set up to track remote branch 'v7r2-fixes33' from 'origin'.
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
```

Reply:
Sorry about that, the correct syntax is:
```bash
pip install git+https://github.com/fstagni/DIRAC.git@v7r2-fixes33#egg=DIRAC[pilot]
```